server/lyft/gateway/autoplan_builder.go

```diff
@@ -0,0 +1,166 @@
+package gateway
+
+import (
+	"fmt"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/metrics"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/vcs"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/recovery"
+	"github.com/uber-go/tally"
+	"strconv"
+)
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_autoplan_validator.go AutoplanValidator
+type AutoplanValidator interface {
+	PullRequestHasTerraformChanges(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool
+}
+
+// AutoplanBuilder handles setting up repo cloning and checking to verify of any terraform files have changed
+type AutoplanBuilder struct {
+	Logger                        logging.SimpleLogging
+	Scope                         tally.Scope
+	VCSClient                     vcs.Client
+	PreWorkflowHooksCommandRunner events.PreWorkflowHooksCommandRunner
+	DisableAutoplan               bool
+	Drainer                       *events.Drainer
+	GlobalCfg                     valid.GlobalCfg
+	// AllowForkPRs controls whether we operate on pull requests from forks.
+	AllowForkPRs bool
+	// AllowForkPRsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to enable this functionality.
+	AllowForkPRsFlag string
+	// SilenceForkPRErrors controls whether to comment on Fork PRs when AllowForkPRs = False
+	SilenceForkPRErrors bool
+	// SilenceForkPRErrorsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to disable error comment
+	SilenceForkPRErrorsFlag string
+	// SilenceVCSStatusNoPlans is whether autoplan should set commit status if no plans
+	// are found
+	silenceVCSStatusNoPlans bool
+	// SilenceVCSStatusNoPlans is whether any plan should set commit status if no projects
+	// are found
+	silenceVCSStatusNoProjects bool
+	CommitStatusUpdater        events.CommitStatusUpdater
+	PrjCmdBuilder              events.ProjectPlanCommandBuilder
+	PullUpdater                *events.PullUpdater
+}
+
+func (r *AutoplanBuilder) PullRequestHasTerraformChanges(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool {
```

Comment:
This naming is oddly specific. Would be more open to `AutoplanValidator` and the function would be `isValid` or something like: `EventFilter` and the func would be `ShouldFilter`